### ISSUE TITLE
Efficiency is much worse when data_sample_strategy is GOSS

### ISSUE THREAD

**Issue by AllenSun1024** [microsoft/LightGBM#5830]
## Description
With the same parameters except **data_sample_strategy**, time elapsed for training with parameter `data_sample_strategy = bagging` is 194s while that with parameter `data_sample_strategy = goss` is 1754s, which is 10x slower approximately.

As stated in paper <<LightGBM: A Highly Efficient Gradient Boosting Decision Tree>>, **GOSS can obtain quite accurate estimation of the information gain with a much smaller data size**. However, it doesn't perform well in reality. 

Why?

**Comment by guolinke** [comment]
there are several possible reasons.
1. the speed-up is data-dependent, it is not possible to have the same speed-up for all kinds datasets. 
2. Parameters, like early stopping may result in very different speeds.
3. Versions. There are significant changes after the paper, some changes may affect the speed. you can try the early version of LightGBM.

**Comment by AllenSun1024** [comment]
> there are several possible reasons.
> 
> 1. the speed-up is data-dependent, it is not possible to have the same speed-up for all kinds datasets.
> 2. Parameters, like early stopping may result in very different speeds.
> 3. Versions. There are significant changes after the paper, some changes may affect the speed. you can try the early version of LightGBM.

Got it, thank you anyway.
